packages/slate/src/transforms/node.ts

```diff
@@ -67,7 +68,7 @@ export interface NodeTransforms {
   ) => void
   setNodes: <T extends Node>(
     editor: Editor,
-    props: Partial<Node>,
+    props: Partial<T> | ((existingProps: NodeProps) => Partial<T>),
```

Comment:
I would like to change the type of `NodeProps` to be generic.  Can you try the below and let me know if it works?

`type NodeProps<T extends Node> = T extends Element ? Omit<T, 'children'> : Omit<T, 'text'>;`

Reply:
I'll change this in the interfaces/node.ts file right?

Reply:
Wherever the type is defined change it there.  Thx.